### DISCUSSION TITLE
Pause time tracking when screen is locked

### DISCUSSION THREAD

**Discussion by aircodepl** [super-productivity/super-productivity#3398] [Ideas]
Hi,
Firstly, I want to thank you for your updates. Recent changes transformed Super Productivity into my go-to life/work task manager.

There is only one feature missing, which makes Time Tracking a bit unreliable for me: Lock screen / idle handler. Seems nobody mentioned it and I cannot see proper options available.

In the TopTracker there is a function that stops Time Tracker when:
- you are idle for X minutes,
- your screen is locked.

I'm usually working with settings like:
- Time Tracker: 5 minute pomodoro break setting,
- Ubuntu: lock screen after 10 minutes.

This means that my pomodoro breaks are part of a job and are counted into total time, but if I go away for 10 minutes the screen will lock (or Time Tracker will see that I was idle for 10 minutes) and tracker should pause. It will unpause if I log back again.

**What is needed?**

In the `Settings` -> `Time Tracking` there should be options:

- Pause time tracking on screen lock,
- Pause time tracking after idle time (in minutes), _(it needs to take pomodoro breaks into account)._

idle time = when user is not using either mouse or keyboard

**Current behavior**

If I start time tracking and go away for 3 hours because something happened, after logging back to system I will have 3 hours tracked. This may be expected for some people.

**Expected behavior**

Tracker should (or not) be paused when screen lock or idle time limit happens. When user come back, it should start tracking from the moment it stopped (no time added).

**Comment by UsamaKarim** [comment]
I opened the issue yesterday about it. #3396

I agree the recent update (v10) is really great and pushed a lot ahead in terms of usability. @johannesjo is doing really amazing job. I am using this tool for 3 weeks, and it feels like this tools is especially curated for programmers.

> In the Settings -> Time Tracking there should be options:
>
>Pause time tracking on screen lock,
>Pause time tracking after idle time (in minutes), (it needs to take pomodoro breaks into account).

If you have auto lock screen timer, for example after 10 minutes, the only feature that needs to implemented is stopping tracking the time on lock screen. As the auto lock screen will eventually stop the timer and idle timer will be started.

**Reply by aircodepl** [reply]
I'm sorry. I was looking for this issue and was blind or something. I'll close this discussion and just bump up your issue :).
